File: src/main/java/com/hotakademy/dockwitcher/domain/repositories/IMongoRepository.java
package com.hotakademy.dockwitcher.domain.repositories;

import com.hotakademy.dockwitcher.domain.entities.Entity;
import org.bson.types.ObjectId;

import java.util.Optional;

public interface IMongoRepository<TEntity extends Entity> {
    void save(TEntity entity);

    Optional<TEntity> findById(ObjectId objectId);

    boolean existsById(ObjectId id);
}

File: src/main/kotlin/com/hotakakademy/dockwitcher/domain/repositories/IMongoRepository.kt
package com.hotakakademy.dockwitcher.domain.repositories

import com.hotakakademy.dockwitcher.domain.entities.Entity
import org.bson.types.ObjectId

interface IMongoRepository<TEntity : Entity> {
    fun save(entity: TEntity)
    fun findById(objectId: ObjectId?): TEntity?
    fun existsById(id: ObjectId?): Boolean
}